test/unit/mdc-toolbar/foundation.test.js

```diff
@@ -0,0 +1,685 @@
+  foundation.updateAdjustElementStyles();
+
+  td.verify(mockAdapter.setStyleForFixedAdjustElement('marginTop', td.matchers.anything()));
+});
+
+test('#updateAdjustElementStyles do not adjust marginTop for non-fixed toolbar', () => {
+  const {foundation, mockAdapter} = setupTest();
+
+  foundation.init();
+  foundation.updateAdjustElementStyles();
+
+  td.verify(mockAdapter.setStyleForFixedAdjustElement('marginTop', td.matchers.anything()), {times: 0});
+});
+
+test('#updateToolbarStyles debounces calls within the same frame', () => {
+  const {foundation} = setupTest();
+  const mockRaf = createMockRaf();
+  foundation.updateToolbarStyles();
+  foundation.updateToolbarStyles();
+  foundation.updateToolbarStyles();
+  assert.equal(mockRaf.pendingFrames.length, 1);
+  mockRaf.restore();
+});
+
+test('#updateToolbarStyles resets debounce latch when updateToolbarStyles frame is run', () => {
+  const {foundation} = setupTest();
+  const mockRaf = createMockRaf();
+
+  foundation.updateToolbarStyles();
+  mockRaf.flush();
+  foundation.updateToolbarStyles();
+  assert.equal(mockRaf.pendingFrames.length, 1);
+  mockRaf.restore();
+});
+
+test('#updateToolbarStyles will not execute if we scrolled out of theshold and executed last change', () => {
+  const {foundation, mockAdapter} = setupTest();
+  const mockRaf = createMockRaf();
+
+  foundation.scrolledOutOfTheshold_ = () => true;
+  foundation.executedLastChange_ = true;
+  foundation.updateToolbarStyles();
+  mockRaf.flush();
+
+  td.verify(mockAdapter.notifyChange({flexibleExpansionRatio: td.matchers.isA(Number)}), {times: 0});
+  mockRaf.restore();
+});
+
+test('#updateToolbarStyles will execute if we scrolled out but last change was not executed, then set it to true',
+  () => {
+  const {foundation, mockAdapter} = setupTest();
+  const mockRaf = createMockRaf();
+
+  foundation.scrolledOutOfTheshold_ = () => true;
+  foundation.updateToolbarStyles();
+  mockRaf.flush();
+
+  td.verify(mockAdapter.notifyChange({flexibleExpansionRatio: td.matchers.isA(Number)}));
+  assert.equal(foundation.executedLastChange_, true);
+  mockRaf.restore();
+});
+
+test('#updateToolbarStyles will execute if we have not scrolled out of theshold, keep executedLastChange_ false',
+  () => {
+  const {foundation, mockAdapter} = setupTest();
+  const mockRaf = createMockRaf();
+
+  foundation.scrolledOutOfTheshold_ = () => false;
+  foundation.updateToolbarStyles();
+  mockRaf.flush();
+
+  td.verify(mockAdapter.notifyChange({flexibleExpansionRatio: td.matchers.isA(Number)}));
+  assert.equal(foundation.executedLastChange_, false);
+  mockRaf.restore();
+});
+
+test('#updateToolbarStyles take correct action for scrollable flexible header when flexible space fully expaned',
+  () => {
+  const {foundation, mockAdapter} = setupTest();
+  const mockRaf = createMockRaf();
+
+  foundation.hasFlexibleRow_ = true;
+  foundation.scrolledOutOfTheshold_ = () => false;
+  foundation.getFlexibleExpansionRatio = () => 1;
```

Comment:
We should never be mocking out methods on a module under test, because then the module isn't really being tested because its behavior is not reflective of reality.

Similar to above, prefer to simulate the interactions that would cause flexibleExpansionRatio to equal `1` and then test the methods